chia/_tests/wallet/test_clvm_casts.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+import random
+
+import pytest
+
+from chia.types.blockchain_format.program import Program
+from chia.util.casts import int_from_bytes, int_to_bytes
+
+
+@pytest.mark.parametrize(
+    "value",
+    [
+        0,
+        1,
+        -1,
+        127,
+        -127,
+        128,
+        -128,
+        255,
+        -255,
+        256,
+        -256,
+        2**8 - 1,
+        -(2**8) + 1,
+        2**15,
+        -(2**15),
+        2**31 - 1,
+        -(2**31),
+        2**63 - 1,
+        -(2**63),
+        2**127 - 1,
+        -(2**127),
+        10**100,
+        -(10**100),  # big ints
+    ],
+)
+def test_round_trip(value: int) -> None:
+    t1 = Program.to(value)
+    t2 = int_to_bytes(value)
+    assert t1 == Program.to(t2)
+    assert int_from_bytes(t2) == value
+    assert t1.as_int() == value
+
+
+def test_zero_serialization() -> None:
+    assert int_to_bytes(0) == b""
+    assert int_from_bytes(b"") == 0
+
+
+def test_minimal_encoding() -> None:
+    assert int_to_bytes(0x80) == b"\x00\x80"
+    assert int_to_bytes(-129) == b"\xff\x7f"  # no extra FF
+    assert int_to_bytes(-128) == b"\x80"
+    assert int_to_bytes(127) == b"\x7f"
+
+
+def test_randomized() -> None:
+    for _ in range(1000):
+        v = random.randint(-(2**256), 2**256)
```

Comment:
it's preferable to have unit tests be deterministic, so this would be better using fixed seed